### PR TITLE
enable tailwind nesting

### PR DESCRIPTION
**Pull request by rafh**
Currently, if you implement native CSS nesting within a Vue component a warning will appear in the terminal. It states 
`Nested CSS was detected, but CSS nesting has not been configured correctly.
Please enable a CSS nesting plugin *before* Tailwind in your configuration.` To fix this error we need to enable the built-in [tailwinds nesting config](https://tailwindcss.com/docs/using-with-preprocessors#nesting).

Example code to trigger the warning within a vue component:

```CSS
<style>
.example {
  &:hover,
  &:focus-visible {
    color: var(--color-text);
  }

  & svg {
    margin-right: 0.78rem;
  }
}
</style>
```

### CODE REVIEW

**Comment by silverwind** [comment]
Won't this also get us [postcss-nested](https://github.com/postcss/postcss-nested), which is a nonstandard nesting syntax I would like to avoid.

**Comment by jtran** [comment]
We already have `postcss-nested` in our dependencies.

https://github.com/go-gitea/gitea/blob/857243bed7f9dccdc597c2941df41e821781cb0f/package-lock.json#L11504

@silverwind, would you prefer to use `postcss-nesting` instead which uses the standard?

**Comment by silverwind** [comment]
Yeah, it's a dependency of tailwindcss, but my preference would be standard nesting syntax, either not transpiled at all, or transpiled to flat syntax until PaleMoon gains support.

**Comment by silverwind** [comment]
Seems https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-nesting, so please add that as dependency and make it use that instead of `postcss-nested`.

**Review by jtran — COMMENTED** [review]
Looks good to me.

File: package.json [hunk 1]
```diff
@@ -86,6 +86,7 @@
     "jsdom": "24.0.0",
     "markdownlint-cli": "0.39.0",
     "postcss-html": "1.6.0",
+    "postcss-nesting": "12.1.0",
```
Comment (silverwind) [inline package.json]: Move it it to `dependencies` please, everything that the build depends on should be there:

```
npm uninstall postcss-nesting && npm i postcss-nesting

**Comment by silverwind** [comment]
Yes looks good. I will also give it a test later.

**Comment by silverwind** [comment]
One final question: What is the reason for this PR? We don't have any CSS nesting in the code yet as far as I'm aware. Is it for customization or are you preparing a PR that will use it?

**Comment by rafh** [comment]
> One final question: What is the motivation for this PR? We don't have any CSS nesting in the code yet as far as I'm aware. Is it for customization or are you preparing a PR that will use it?

We use CSS nesting within our fork. I thought this would be a good candidate to push upstream because others may eventually use nesting in their PRs.

**Comment by silverwind** [comment]
Ok, personally I wouldn't use it, but we can give people this option.

**Review by jtran — COMMENTED** [review]
👍

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Tested it: In Vue components I see the CSS is flattened as expected. But if I edit a regular CSS file it isn't:

image

Not sure what's the issue, but I would expect CSS to be flattened in both cases.

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
I plan to remove this `postcss-nesting` dependency again in https://github.com/go-gitea/gitea/pull/35508. I think there is no reason to have a dependency we don't use and if you are in a fork of gitea, you ought to control the dependencies yourselfs.

**Pull request merged**